webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx

```diff
@@ -0,0 +1,563 @@
+  colorMap: MapOfIdsToColors;
+  data: ExperimentItem[];
+
+  handleScroll?: (r: Rectangle) => void;
+
+  sortableColumnIds: ExperimentColumn[];
+  setSortableColumnIds: Dispatch<SetStateAction<ExperimentColumn[]>>;
+
+  selectedExperimentIds: string[];
+  setSelectedExperimentIds: Dispatch<SetStateAction<string[]>>;
+  selectAll: boolean;
+  setSelectAll: Dispatch<SetStateAction<boolean>>;
+}
+
+const STATIC_COLUMNS: ExperimentColumn[] = ['selected', 'name'];
+
+type ColumnDef = SizedGridColumn & {
+  id: ExperimentColumn;
+  isNumerical?: boolean;
+  renderer: (record: ExperimentItem, idx: number) => GridCell;
+};
+
+export const GlideTable: React.FC<Props> = ({
+  data,
+  setSelectedExperimentIds,
+  sortableColumnIds,
+  setSortableColumnIds,
+  colorMap,
+  selectAll,
+  setSelectAll,
+  handleScroll,
+}) => {
+  const gridRef = useRef(null);
+
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
+
+  const handleMenuClose = useCallback(() => {
+    setMenuIsOpen(false);
+  }, []);
+
+  const {
+    ui: { darkLight },
+  } = useUI();
+
+  const [menuProps, setMenuProps] = useState<Omit<TableActionMenuProps, 'open'>>({
+    handleClose: handleMenuClose,
+    x: 0,
+    y: 0,
+  });
+
+  const users = Loadable.map(useObservable(usersStore.getUsers()), ({ users }) => users);
+
+  const columnIds = useMemo<ExperimentColumn[]>(
+    () => [...STATIC_COLUMNS, ...sortableColumnIds],
+    [sortableColumnIds],
+  );
+  const navigate = useNavigate();
+  const bodyStyles = getComputedStyle(document.body);
+
+  const [selection, setSelection] = React.useState<GridSelection>({
+    columns: CompactSelection.empty(),
+    rows: CompactSelection.empty(),
+  });
+  const getRowThemeOverride = React.useCallback(
+    (row: number): Partial<Theme> | undefined =>
+      colorMap[data[row]?.id]
+        ? {
+            accentColor: colorMap[data[row]?.id],
+            borderColor: '#F0F0F0',
+          }
+        : { borderColor: '#F0F0F0' },
+    [colorMap, data],
+  );
+
+  useEffect(() => {
+    const selectedRowIndices = selection.rows.toArray();
+    setSelectedExperimentIds((prevIds) => {
+      const selectedIds = selectedRowIndices
+        .map((idx) => String(data?.[idx]?.id))
+        .filter((x) => x !== undefined);
+      if (prevIds === selectedIds) return prevIds;
+      return selectedIds;
+    });
+  }, [selection.rows, setSelectedExperimentIds, data]);
+
+  const theme = getTheme(bodyStyles);
+
+  const [columnWidths, setColumnWidths] = useState<Record<ExperimentColumn, number>>({
+    archived: 80,
+    checkpointCount: 74,
+    checkpointSize: 74,
+    description: 148,
+    duration: 96,
+    forkedFrom: 128,
+    id: 50,
+    name: 150,
+    numTrials: 74,
+    progress: 111,
+    resourcePool: 140,
+    searcherMetricValue: 74,
+    searcherType: 140,
+    selected: 45,
+    startTime: 118,
+    state: 106,
+    tags: 106,
+    user: 85,
+  });
+
+  const columnDefs = useMemo<Record<ExperimentColumn, ColumnDef>>(
+    () => ({
```

Comment:
if its possible to change text alignment, numeric col: align right, icon: align center, and string: aligns left 

<img width="401" alt="image" src="https://user-images.githubusercontent.com/109113616/228913809-2d5049a2-3e92-4d89-a7f1-fde5a33350d3.png">